src/test_framework.rs

```diff
@@ -188,10 +188,6 @@ impl TestFramework {
         *self.shutdown.write().unwrap() = true;
         // stop bitcoind
         let _ = self.bitcoind.client.stop().unwrap();
-        // Remove test temp dir, ignore error.
-        if fs::remove_dir_all::<PathBuf>(self.temp_dir.clone()).is_err() {
-            // Do Nothing
-        }
```

Comment:
Why? This will pollute the test datadir with run files. 

Maybe take this out as an independent `clean()` method? Then, it can be selectively commented out in tests to persist files if needed for a test.

but this seems unrelated to the logging problem. So maybe in a separate PR?

Reply:
Yeah I think this should be the right behavior. What we have currently is wrong. `stop` and `clear` should not be done in a single method.

Reply:
wrote it in an issue.
https://github.com/utxo-teleport/teleport-transactions/issues/86

Reply:
What about making temporary directory under `/tmp`?
Something like `/tmp/teleport/tests`. Then we don't have to worry about cleaning as they'll get purged on every reboot.
But I can see some downsides to it, such as what if someone do not want to wait until next reboot and is testing again and again. This will probably just clutter the space.

And this will require a similar approach for others OSes like Windows for example as `/tmp` is standard for only Unix based OSes.

Reply:
@rajarshimaitra we did discussed this in discord. But ya this clean and stop method also works.

Reply:
> What about making temporary directory under /tmp?

Thats more cleaner. I don't remember why I haven't done that. Worth a PR.

Reply:
> @rajarshimaitra we did discussed this in discord. But ya this clean and stop method also works.

I forgot what was the discussion. At first glance, I don't see why we would want to remove this line.